tests/testthat/test-recall_data.R

```diff
@@ -0,0 +1,206 @@
+test_that("recall_by_make returns the correct shape",{
+    result <- recall_by_make('Nissan')
+    expect_type(result, "list")
+    expect_equal(nrow(result), 25)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_by_make by year range has the correct shape",{
+    result <- recall_by_make('Nissan', start_year = 1980, end_year=2005)
+    expect_gt(nrow(result), 0)
+    filtered <- subset(result, Year >= 1980 & Year <= 2005)
+    expect_equal(nrow(result), nrow(filtered))
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+    result <- recall_by_make('Nissan', start_year = 1980)
+    expect_gt(nrow(result), 0)
+    result <- recall_by_make('Nissan', end_year = 1980)
+    expect_gt(nrow(result), 0)
+
+})
+
+test_that("recall_by_make using manufacturer has the correct shape",{
+    result <- recall_by_make('Nissan', manufacturer = TRUE)
+    expect_equal(nrow(result), 25)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+    result <- recall_by_make('Nissan', start_year = 1980)
+    expect_gt(nrow(result), 0)
+    result <- recall_by_make('Nissan', end_year = 1980)
```

Comment:
can probably drop a couple of tests by removing a few of the `manufacturer=TRUE` ones, one call is probably sufficent